packages/gatsby-theme-carbon/src/components/PageTabs/PageTabs.js

```diff
@@ -15,15 +15,21 @@ import {
 export default class PageTabs extends React.Component {
   render() {
     const { tabs, slug } = this.props;
-    const currentTab = slug.split('/').filter(Boolean).slice(-1)[0];
+    const newSlug = slug.replace(/[’']/g, ''); // removes any apostrophes from slug
+    const currentTab = newSlug.split('/').filter(Boolean).slice(-1)[0];
 
     const pageTabs = tabs.map((tab) => {
-      const slugifiedTab = slugify(tab, { lower: true });
+      const tempTab = tab.replace(/[’']/g, ''); // removes any apostrophes from tab
+      const slugifiedTab = slugify(tempTab, { lower: true });
```

Comment:
Slugify has a [`strict` option](https://www.npmjs.com/package/slugify#options) that removes any punctuation/special characters

From a quick test, it strips out the apostrophe so it would match the yaml/filename after processing. I think that should remove the need for the temp tab all together maybe.